OutOfSchool/OutOfSchool.DataAccess/Repository/EntityRepositoryBase.cs

```diff
@@ -161,6 +161,19 @@ public virtual async Task<IEnumerable<TEntity>> GetAllWithDetails(string include
     /// <inheritdoc/>
     public virtual Task<TEntity> GetById(TKey id) => dbSet.FirstOrDefaultAsync(x => x.Id.Equals(id));
 
+    /// <inheritdoc/>
+    public virtual Task<TEntity> GetByIdWithDetails(TKey id, string includeProperties = "")
+    {
+        var query = dbSet.Where(x => x.Id.Equals(id));
+        foreach (var includeProperty in includeProperties.Split(
```

Comment:
(Code style) I understand, that this code was copied from other implementation, thus have few questions. 
1. What if `includeProperties == null`? 
2. Why not extract this `foreach` block into own method?

Reply:
Same for `EntityRepositorySoftDeleted`

Reply:
Fixed. Thank you!

Reply:
I have implemented these improvements and taken note of your recommendations. Thank you!

Reply:
(Code style, optional) And now you could do one last touch of "perfection" ))

In all places where `IncludeProperties()` is used you could write "oneliners". 
So instead of
```
        IQueryable<TEntity> query = dbSet;
        query = query.IncludeProperties(includeProperties);
        return await query.ToListAsync();
```
it could be
```
        => dbSet.IncludeProperties(includeProperties).ToListAsync();        
```
Consider removal of excessive `await`.

Reply:
Definitely! As documentation says about 2nd parameter:
"The name of the parameter with which argument corresponds. If you omit this parameter, the name of argument is used."
So that's better to have less code and make Sonar happy

Reply:
I agree that wording "last touch" was too optimistic. 

As suggested change induces or is related to 

- possible usage of cancellation tokens during EF queries, 
- changes of repo methods' return types to `IAsyncEnumerable<TEntity>` (and thus necessity of `System.Linq.Async` package) or `Tas<List<TEntity>>` that is more common, though not always performance friendly

it is easier to make it in separate PR. 

@DKymachynskyi , please, create a tech debt issue with information from this and beforementioned comment

Reply:
I have created the issue. Thank you!